lib/PubNub/PubSub/Message.pm

```diff
@@ -66,10 +67,10 @@ sub payload {
 Returns a message object with a payload from a json string.
 
 =cut
-
+my $json = JSON::MaybeXS->new->allow_nonref(1);
 sub from_msg {
     my ($self, $json) = @_;
```

Comment:
If this is a class method (as suggested by the oddly-stringified `"$self"` later on), this wants to be

    my ($class, $json) = @_;